src/AudioToolbox/MusicSequence.cs

```diff
@@ -274,6 +288,39 @@ public double GetBeatsForSeconds (double seconds)
 				return beats;
 			return 0;
 		}
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicSequenceSetUserCallback (/* MusicSequence */ IntPtr inSequence, MusicSequenceUserCallbackProxy inCallback, /* void * */ IntPtr inClientData);
+
+		public void SetUserCallback (MusicSequenceUserCallback callback)
+		{
+			lock (userCallbackHandles) {
+				if (userCallbackHandles.ContainsKey (handle))
+					userCallbackHandles[handle] = callback;
```

Comment:
Using the indexer will work both when you're adding a new entry as well as when modifying an existing entry, so the ContainsKey check is redundant.